src/rgw/rgw_sal_motr.cc

```diff
@@ -3624,7 +3624,8 @@ int MotrStore::next_query_by_name(string idx_name,
       ++k;
     }
 
-    if (rc < (int)nr_kvp) // there are no more keys to fetch
+    int keys_left = val_out.size() - (i+k);  // i+k gives next index.
+    if (rc < (int)nr_kvp || keys_left <= 0)  // No more keys to fetch
```

Comment:
The check for keys_left will be done in the for loop above at line 3596, won't it?

Reply:
I am checking for keys_left before fetching the keys itself from 2nd iteration. because in the first iteration atleast the issue can not occure.

Reply:
I mean the check for the 2nd and following fetch iterations too.

Reply:
removed an extra check. 👍